js/src/Components/Widgets/LinkWidget.js

```diff
@@ -118,48 +135,95 @@ export const LinkWidget = ({fieldId, defaultValue, onFieldChange, settings}) =>
   /**
    * A suggestion was picked, pass that up to the manager to store the value.
    */
-  const suggestionPicked = (e, selectedValue) => {
-    onFieldChange([{
-      title: defaultFieldValue.title,
-      uri: getUriFromString(selectedValue === null ? '' : selectedValue.value)
-    }]);
+  const suggestionPicked = (e, selectedValue, delta) => {
+    alterValues({
+      title: fieldValues[delta].title,
+      uri: getUriFromString(selectedValue === null ? '' : selectedValue.value),
+      delta: delta
+    });
   };
 
   /**
    * When the textfield on the URI blurs, that's when we want to trigger the field change and pass it up to the manager.
    */
-  const onUriBlur = (e) => {
-    onFieldChange([{
-      title: defaultFieldValue.title,
-      uri: getUriFromString(e.target.value)
-    }]);
+  const onUriBlur = (e, delta) => {
+    alterValues({
+      title: fieldValues[delta].title,
+      uri: getUriFromString(e.target.value),
+      delta: delta
+    });
   }
 
-  return (
-    <FormGroup>
-      <FormLabel component="legend">
-        {settings.label}
-      </FormLabel>
+  /**
+   * To support cardinality, we will need an "Add another" button.
+   */
+  const addAnotherButton = () => {
+    if ((settings.cardinality == -1) || (settings.cardinality > fieldValues.length )) {
+      return (
+      <div>
+        <Button variant="outlined" style={{marginBottom: '20px'}} onClick={addAnother}>
+          Add Another Link
+        </Button>
+      </div>
+      );
+    }
+  }
+
+  /**
+   * Handler for the addAnotherButton
+   */
+  const addAnother = () => {
+    const newState = fieldValues.concat(emptyLinkValue);
+    setValues(newState);
+  }
 
-      <FormControl style={{marginBottom: '20px'}}>
+  /**
+   * If we add more links, we need a way of removing them, but never remove link 0.
+   */
+  const removeLinkButton = (delta) => {
+    if (delta > 0){
```

Comment:
It would be nice to be able to delete the first link as well.

Reply:
Fixed.  In order to prevent re-draws with an empty array, I have it add a new, empty link if the count of the array reaches zero.  That should prevent render bugs if all the links are deleted.